pymodbus/server/async_io.py

```diff
@@ -574,7 +572,7 @@ def __init__(
             True,
         )
 
-        self.loop = kwargs.get("loop") or asyncio.get_event_loop()
+        self.loop = asyncio.get_event_loop()
```

Comment:
please use get_running_loop

Reply:
When I do this, pymodbus.server doesn't start with the following error message:
```
pymodbus.server --verbose run -s serial -p ./device/p1 -f rtu
Traceback (most recent call last):

  File "/usr/bin/pymodbus.server", line 8, in <module>
    sys.exit(app())
             ^^^^^

  File "/usr/lib/python3.11/site-packages/pymodbus/repl/server/main.py", line 183, in run
    app = ReactiveServer.factory(
          ^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/pymodbus/server/reactive/main.py", line 498, in factory
    server = server(context, framer=framer, identity=identity, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/lib/python3.11/site-packages/pymodbus/server/async_io.py", line 575, in __init__
    self.loop = asyncio.get_running_loop()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^

RuntimeError: no running event loop
```
I'm afraid I don't fully understand the design here - especially which class is responsible for creating the event loop...

Reply:
I am afraid there are no bigger design, but merely a not well maintained corner.

main needs to start the app with asyncio.run, that way the loop will exist. Which might need to be done in 2 steps using an async helper method because app is not async.

asyncio.py contains convenience functions to do that (startXserver), but repl uses some old structure.

Reply:
> please use get_running_loop

Fixed in latest commits.